[MM-57209] Set EmailVerified to true when promoting synthetic users

JulienTant opened the pull request:
#### Summary
Set EmailVerified to true when promoting a synthetic user. 	

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57209

lieut-data reviewed (changes requested):
Thank you, @JulienTant! Can you also make these changes in the `syncJob` that's in `plugin.go`? This is where we first create synthetic users, and where we'll need to support updating these in bulk as well, as some users are promoted automatically without the explicit slash command.

JulienTant commented:
Thanks, @lieut-data! I applied the changes to the sync users method.

Applying those changes, and adding tests, made me realize that the tests for this method were not working because the MSUser was created with a `Mail` field, which would cause the main loop to `continue` without doing anything (first if of the loop). I added the `Mail` field and fixed the mock calls as needed + added 2 tests for user creation and update.

lieut-data commented on server/testutils/data.go:
```diff
@@ -102,6 +102,16 @@ func GetUser(role, email string) *model.User {
+		Username: "test-user",
+		Roles:    role,
+		Email:    email,
+		RemoteId: &remoteID,
```
For completeness, should we set `EmailVerified` to true for this test user?

JulienTant replied:
Hum, I purposely left it to `false` to make sure my code set it to `true` 😅

lieut-data replied:
No, that's fine too! Just wanted to clarify -- your call either way :)

lieut-data reviewed (approved):
Thanks, @JulienTant!

JulienTant commented:
/update-branch

lindy65 reviewed (approved):
Thanks @JulienTant - this will be tested post-merge, QA-deferring.
